### PR TITLE
Add Google C# style guide

### PR DESCRIPTION
**Pull request by jbcoe**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +37,7 @@ The following Google style guides live outside of this project:
 <a rel="license" href="https://creativecommons.org/licenses/by/3.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/3.0/88x31.png" /></a>
 
 [cpp]: https://google.github.io/styleguide/cppguide.html
+[csharp]: https://google.github.io/styleguide/csharp-style.html
```
Comment (tituswinters) [inline README.md:40]: This link is .html and the other file is .md

File: csharp-style.md [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+#### Code
+
+*   Names of classes, methods, enumerations, public fields, public properties,
+    namespaces: `PascalCase`.
+*   Names of local variables, parameters: `camelCase`.
+*   Names of private, protected, internal and protected internal fields and
+    properties: `_camelCase`.
+*   Naming convention is unaffected by modifiers such as const, static,
+    readonly, etc.
+*   For casing, a "word" is anything written without internal spaces, including
```
Comment (tituswinters) [inline csharp-style.md:28]: I don't have any authority here, but I suspect this rule is a lost cause.

File: csharp-style.md [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+*   Where possible the file name should be the same as the name of the main
+    class in the file, e.g. `MyClass.cs`.
+*   In general, prefer one core class per file.
+
+### Organization
+
+*   Modifiers occur in the following order: `public protected internal private
+    new abstract virtual override sealed static readonly extern unsafe volatile
+    async`.
+*   Namespace `using` declarations go at the top, before any namespaces. 'using'
```
Comment (tituswinters) [inline csharp-style.md]: Should the second using be in backticks?

File: csharp-style.md [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+    *   For transient variables that are only passed directly to other methods -
+        e.g. `var item = GetItem(); ProcessItem(item);`
+
+*   Discouraged:
+
+    *   When working with basic types - e.g. `var success = true;`
+    *   When working with compiler-resolved built-in numeric types - e.g. `var
+        number = 12 * ReturnsFloat();`
+    *   When users would clearly benefit from knowing the type - e.g. `var
+        listOfItems = GetList();
```
Comment (tituswinters) [inline csharp-style.md]: missing a close `

File: README.md (hunk 1 above)
Comment (jbcoe) [reply README.md:40]: The R guide does this as well and seems to work - must be some GitHub magic.

Can amend or not as you like.

File: csharp-style.md (hunk 4 above)
Comment (jbcoe) [reply csharp-style.md]: Fixed and updated.

File: csharp-style.md (hunk 3 above)
Comment (jbcoe) [reply csharp-style.md]: Fixed and updated.

File: csharp-style.md (hunk 2 above)
Comment (jbcoe) [reply csharp-style.md:28]: We may update this in time if it proves unenforceable. What's your concern?

Comment (tituswinters) [reply csharp-style.md:28]: C++ has this rule and it's violated all over. Partly just due to "language is squishy".

**Pull request merged**